### DISCUSSION TITLE
Feature Request: password-based file encryption using AES

### DISCUSSION THREAD

**Discussion by jnoel-hrs** [krzyzanowskim/CryptoSwift#1030] [Q&A]
My company needs a mechanism to stream data to local file storage, encrypting the data as it comes in (data comes in sporadically from several different sources over time). I took a stab at implementing this here: [AESEncryptedFile.swift](https://github.com/jnoel-hrs/CryptoSwift/blob/feat/aes-encrypted-file/Sources/CryptoSwift/AESEncryptedFile.swift)

The goal is basically to have something similar to the AndroidX [EncryptedFile API](https://developer.android.com/reference/androidx/security/crypto/EncryptedFile), which is a turn-key solution for encrypting data via streams.

However, my above implementation is failing in the [unit test](https://github.com/jnoel-hrs/CryptoSwift/blob/feat/aes-encrypted-file/Tests/CryptoSwiftTests/AESEncryptedFileTests.swift) I wrote because the data seems to be either getting cut off, or is all NULL bytes.

Any help to fix would be greatly appreciated, and if this looks like a good addition I'll happily open a PR for it.

Thanks!

**Comment by jnoel-hrs** [comment]
Managed to get this working with a different repo [IDZSwiftCommonCrypto](https://github.com/iosdevzone/IDZSwiftCommonCrypto/pull/131) - although I suspect the main issue was that I was relying on `hasBytesAvailable` which apparently is not always reliable.

**Comment by krzyzanowskim** [comment]
😩
